### PR TITLE
AIP re-ingest

### PR DESCRIPTION
**Pull request by Hwesta**
Add re-ingest functionality to the storage service
- Add re-ingest button to package page
- Add endpoint for Archivematica to start a re-ingest
- Extract and copy files to Archivematica to start re-ingest
- Accept a reingested package, updates/replaces METS & metadata files, and stores
- Known bug: Does not support uncompressed AIPs.  Does support .tar.bz2 & .zip

Companion to artefactual/archivematica#257

### CODE REVIEW

File: storage_service/locations/models/package.py [hunk 1]
```diff
@@ -820,6 +829,409 @@ def delete_from_storage(self):
...
+            LOGGER.debug('Extract file RC: %s', rc)
+            # Get output path
+            command = ['lsar', '-ja', reingest_full_path]
+            try:
+                output = subprocess.check_output(command)
+                j = json.loads(output)
+                bname = sorted([d['XADFileName'] for d in j['lsarContents'] if d.get('XADIsDirectory', False)], key=len)[0]
+            except (subprocess.CalledProcessError, ValueError):
+                # FIXME
+                print 'I dunno what to do here'
```
Comment (mistydemeo) [inline storage_service/locations/models/package.py]: :scream_cat:

File: storage_service/locations/models/package.py [hunk 2]
```diff
@@ -820,6 +829,409 @@ def delete_from_storage(self):
...
+                original_metadata_dir)
+
+        # Update bag payload and verify
+        bag = bagit.Bag(path)
+        bag.save(manifests=True)
+        bag.validate()  # Raises exception in case of problem
+
+        # Compress
+        # TODO update this for uncompressed AIPs
+        reingest_root = etree.parse(reingest_pointer)
```
Comment (mistydemeo) [inline storage_service/locations/models/package.py:1048]: Maybe handle the absence of a pointer file as meaning uncompressed AIP?

**Comment by mistydemeo** [comment]
Do we want to keep a copy of the original AIP in case something goes wrong with the reingestion process?

File: storage_service/locations/models/package.py (hunk 2 above)
Comment (Hwesta) [inline storage_service/locations/models/package.py:1048]: A good idea, but I think out of scope for the moment.

**Comment by Hwesta** [comment]
> Do we want to keep a copy of the original AIP in case something goes wrong with the reingestion process?

I'd be more comfortable with that, but there are storage concerns. We don't currently make any changes to the original objects, and the AIP in storage isn't modified until after it's repackaged with new METS & metadata.  Perhaps an AIP validation tool, but I think that's out of scope currently.

File: storage_service/locations/models/package.py (hunk 1 above)
Comment (Hwesta) [inline storage_service/locations/models/package.py]: Fixed with default value and warning log.

**Pull request merged**